scripts/standby_node_health_check/standby_node_health_check.py

```diff
@@ -0,0 +1,530 @@
+        default="INFO",
+        help="Set the logging level.",
+    )
+    parser.add_argument("--version", action="version", version="%(prog)s 1.0.0")
+
+    args = parser.parse_args()
+    return args
+
+
+class StandbyNodeHealthChecker:
+    UFM_HA_CLUSTER_COMMAND = "ufm_ha_cluster {}"
+    SYSTEMCTL_IS_ACTIVE_COMMAND = "systemctl is-active {}.service"
+    IS_HA_COMMAND = UFM_HA_CLUSTER_COMMAND.format("is-ha")
+    IS_MASTER_COMMAND = UFM_HA_CLUSTER_COMMAND.format("is-master")
+    HA_STATUS_COMMAND = UFM_HA_CLUSTER_COMMAND.format("status")
+    IBDEV2NETDEV_COMMAND = "ibdev2netdev"
+    IBSTAT_COMMAND = "ibstat"
+    IP_LINK_SHOW_COMMAND = "ip --json link show"
+    PCS_STATUS_COMMAND = "pcs status"
+    COROSYNC_RINGS_COMMAND = "corosync-cfgtool -s"
+
+    IBDEV2NETDEV_REGEX = re.compile(r"^([\w\d_]+) port \d ==> ([\w\d]+)")
+    OLD_CORSYNC_RING_ID_REGEX = re.compile(r"^RING ID (\d+)")
+    OLD_CORSYNC_RING_IP_REGEX = re.compile(r"id\ *= ([\d\.]+)")
+    OLD_CORSTNC_STATUS_REGEX = re.compile(r"^status(?:=|:)\s*(.*)$")
+
+    NEW_CORSYNC_RING_ID_REGEX = re.compile(r"^LINK ID (\d+)")
+    NEW_CORSYNC_RING_ID_IP_REGEX = re.compile(r"addr\ *= ([\d\.]+)")
+    NEW_CORSYNC_STATUS_REGEX = re.compile(
+        r"^node (\d+):link enabled:(\d)link connected:(\d)"
+    )
+
+    def __init__(self, fabric_interfaces, mgmt_interface):
+        self._fabric_interfaces = fabric_interfaces
+        self._mgmt_interface = mgmt_interface
+        self._ufm_ha_status_string = ""
+        self._summary_actions = []
+
+    @classmethod
+    def _run_command(cls, command: str):
+        """
+        The command is a string and we need to split it into an array"""
+        try:
+            command = command.split(" ")
+            result = subprocess.run(
+                command,
+                shell=False,
+                stdout=subprocess.PIPE,
+                # stderr=subprocess.STDOUT,
+                universal_newlines=True,
+                check=False,
+            )
+            return result.returncode, result.stdout.replace("\t", "").strip()
+        except Exception as e:
+            return -1, "", str(e)
+
+    def _check_ib_interfaces(self):
+        result = True
+        ib_interfaces_status = self._run_and_parse_ibstat()
+        ib_to_ml_map = self._get_ib_to_mlx_port_mapping()
+        for ib_interface in self._fabric_interfaces:
+            ib_interface_to_validate = ib_interface
+            if not ib_interface.startswith("mlx"):
+                ib_interface_to_validate = ib_to_ml_map.get(ib_interface, ib_interface)
+            if ib_interface_to_validate not in ib_interfaces_status:
+                logger.warning("%s is not in the list of IB interfaces", ib_interface)
+                result = False
+            elif not self._check_ib_interface(
+                ib_interface_to_validate, ib_interfaces_status
+            ):
+                logger.warning(
+                    "IB interface %s is not active",
```

Comment:
what happen if all IB interfaces are not active? 
shouldn't we log error message and exit?

Reply:
We decided to run all the checks even if one is failing.
A general warning will be in the summary if any interference is down.